src/storage/src/types/sources.rs

```diff
@@ -1609,7 +1609,7 @@ impl RustType<ProtoS3KeySource> for S3KeySource {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
I added these to make `SourceData` compatible with `consolidate_updates`, which requires the data type to implement `Ord`.